models/scheduled_pull_request_merge.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 Gitea. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import "code.gitea.io/gitea/modules/timeutil"
+
+// ScheduledPullRequestMerge represents a pull request scheduled for merging when checks succeed
+type ScheduledPullRequestMerge struct {
+	ID          int64              `xorm:"pk autoincr"`
+	PullID      int64              `xorm:"BIGINT"`
+	UserID      int64              `xorm:"BIGINT"`
+	User        *User              `xorm:"-"`
+	MergeStyle  MergeStyle         `xorm:"varchar(50)"`
+	Message     string             `xorm:"TEXT"`
+	CreatedUnix timeutil.TimeStamp `xorm:"created"`
+}
+
+// ScheduleAutoMerge schedules a pull request to be merged when all checks succeed
+func ScheduleAutoMerge(opts *ScheduledPullRequestMerge) (err error) {
+	// Check if we already have a merge scheduled for that pull request
+	exists, err := x.Exist(&ScheduledPullRequestMerge{PullID: opts.PullID})
+	if err != nil {
+		return
+	}
+	if exists {
+		return ErrPullRequestAlreadyScheduledToAutoMerge{PullID: opts.PullID}
+	}
+
+	opts.UserID = opts.User.ID
+
+	_, err = x.Insert(opts)
+	if err != nil {
+		return
+	}
+
+	pr, err := GetPullRequestByID(opts.PullID)
+	if err != nil {
+		return err
+	}
+
+	_, err = CreateScheduledPRToAutoMergeComment(opts.User, pr)
+	return err
+}
+
+// GetScheduledMergeRequestByPullID gets a scheduled pull request merge by pull request id
+func GetScheduledMergeRequestByPullID(pullID int64) (exists bool, scheduledPRM *ScheduledPullRequestMerge, err error) {
+	scheduledPRM = &ScheduledPullRequestMerge{}
+	exists, err = x.Where("pull_id = ?", pullID).Get(scheduledPRM)
+	if err != nil || !exists {
+		return
+	}
+	scheduledPRM.User, err = getUserByID(x, scheduledPRM.UserID)
+	return
+}
+
+// RemoveScheduledMergeRequest cancels a previously scheduled pull request
+func RemoveScheduledMergeRequest(scheduledPR *ScheduledPullRequestMerge) (err error) {
+	if scheduledPR.ID == 0 && scheduledPR.PullID != 0 {
```

Comment:
can you add the two dirrerent delete requests in one session?

Reply:
But only one will ever be executed :thinking: 

If the condition in L60 is true, it will only execute the first one and then return. IMHO there's no need to put them both in one db session.

(Unless you mean the function itself should be executed in the same session as the caller one is currently in, but that's a whole different story.)

Reply:
missed the return at the end of the if-block

you can resolve this :)